src/app/profile-edit/profile-edit.page.ts

```diff
@@ -92,12 +101,29 @@ export class ProfileEditPage {
       )
       .subscribe((res: any) => {
         if (res?.status === 200) {
-          const result = control.dynamicEntity ? res?.result : res?.result?.data;
+          let result;
+          if (control.dynamicUrl) {
+            result = res?.result;
+          } else {
+            result = control.dynamicEntity ? res?.result : res?.result?.data;
+          }
+          let options;
           if (result) {
-            const options = result.map((entity: any) => ({
-              label: entity.name,
-              value: entity._id
-            }));
+            if (control.dynamicUrl) {
+              options = result.map((entity: any) => ({
```

Comment:
this logic can be moved up where you're checking if dynamic url is present, move up the else logic as well, it'll reduce cognitive load on the reader and improve readability